.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,10 @@
+- id: mkdocs-build
+  name: mkdocs build
+  description: tests if your documentation builds successfully.
+  language: python
+  entry: mkdocs build
+  require_serial: true
+  pass_filenames: false
+  types_or: [markdown, yaml]
+  files: ((^|/)mkdocs\.ya?ml$)|(^docs/)
```

Comment:
I think that the `files` field should be just rewritten as `mkdocs\.ya?ml$` because you can change the `docs/` directory with [`docs_dir` configuration field](https://www.mkdocs.org/user-guide/configuration/#docs_dir) and the *mkdocs.yaml* file could be in nested subdirectories as well.

Reply:
But based on what @Cube707 explained, this value is just a default value which makes sense in most cases. Users can change it in their projects if needed.

Reply:
But does not provide value at all, it does nothing because `pass_filenames` is `false`. See [`files`](https://pre-commit.com/#config-files).

Reply:
yes what @pawamoy said. But also the `files` field is responsible for limiting what files trigger a `mkdoc build` and if removed it will run for changes to any file. As mkdoc is quite expencive to run it should defnetly be limited in when it runs.

An argument could be made here to match **no** files at all by default, making it that it must be run manually (via `pre-commit run mkdocs-build`) or the user must provide (/overide) the file patterns